publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/NonDegreePermissionStrategyTest.java

```diff
@@ -20,14 +25,52 @@
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.EnumSource;
-import org.junit.jupiter.params.provider.EnumSource.Mode;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class NonDegreePermissionStrategyTest extends PublicationPermissionStrategyTest {
 
-    @ParameterizedTest(name = "Should deny Curator {0} operation on degree resources belonging to the institution")
-    @EnumSource(value = PublicationOperation.class)
-    void shouldDenyCuratorOnDegree(PublicationOperation operation)
+    private static Stream<Arguments> argumentsForDenyingCuratorFromPerformingOperationsOnProtectedDegreeResources() {
+        return Stream.of(
+            Arguments.of(PublicationOperation.UPDATE, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreeMaster.class),
+            Arguments.of(PublicationOperation.UPDATE, DegreePhd.class),
+            Arguments.of(PublicationOperation.DELETE, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.DELETE, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.DELETE, DegreeMaster.class),
+            Arguments.of(PublicationOperation.DELETE, DegreePhd.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreeMaster.class),
+            Arguments.of(PublicationOperation.UNPUBLISH, DegreePhd.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreeMaster.class),
+            Arguments.of(PublicationOperation.TERMINATE, DegreePhd.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeLicentiate.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeBachelor.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreeMaster.class),
+            Arguments.of(PublicationOperation.TICKET_PUBLISH, DegreePhd.class)
+        );
+    }
```

Comment:
This one can be shortened, needs test adjustment.
```
    public static final List<Class<? extends DegreeBase>> DEGREES = List.of(DegreeLicentiate.class,
                                                                            DegreeBachelor.class,
                                                                            DegreeMaster.class,
                                                                            DegreePhd.class);

    private static Stream<Arguments> argumentsForDenyingCuratorFromPerformingOperationsOnProtectedDegreeResources() {
        return Stream.of(
            Arguments.of(PublicationOperation.UPDATE, DEGREES),
            Arguments.of(PublicationOperation.DELETE, DEGREES),
            Arguments.of(PublicationOperation.UNPUBLISH, DEGREES),
            Arguments.of(PublicationOperation.TERMINATE, DEGREES),
            Arguments.of(PublicationOperation.TICKET_PUBLISH, DEGREES)
        );
    }
```

Reply:
Then each iteration of the parameterised test would test multiple cases. I thank that is not something we want.

Reply:
how about something like 
```java
    List<Class> degreeClasses = Arrays.asList(DegreeLicentiate.class, DegreeBachelor.class, DegreeMaster.class, DegreePhd.class);
    List<PublicationOperation> operations = Arrays.asList(PublicationOperation.UPDATE, PublicationOperation.DELETE, PublicationOperation.UNPUBLISH, PublicationOperation.TERMINATE, PublicationOperation.TICKET_PUBLISH);
    
    return operations.stream()
        .flatMap(operation -> degreeClasses.stream()
            .map(degreeClass -> Arguments.of(operation, degreeClass)))
        .collect(Collectors.toList())
        .stream();
```